ext/reflection/php_reflection.c

```diff
@@ -2105,7 +2130,124 @@ ZEND_METHOD(ReflectionFunctionAbstract, getNumberOfRequiredParameters)
 }
 /* }}} */
 
-/* {{{ Returns an array of parameter objects for this function */
+/* {{{ Returns whether a parameter exists or not */
+ZEND_METHOD(ReflectionFunctionAbstract, hasParameter)
+{
+    reflection_object *intern;
+    zend_function *fptr;
+    zend_string *arg_name = NULL;
+    zend_long position;
+    uint32_t num_args;
+
+    ZEND_PARSE_PARAMETERS_START(1, 1)
+        Z_PARAM_STR_OR_LONG(arg_name, position)
+    ZEND_PARSE_PARAMETERS_END();
+
+    GET_REFLECTION_OBJECT_PTR(fptr);
+
+    num_args = fptr->common.num_args;
+
+    if (fptr->common.fn_flags & ZEND_ACC_VARIADIC) {
+        num_args++;
+    }
+
+    if (!num_args) {
+        RETURN_FALSE;
+    }
+
+    if (arg_name != NULL) {
+        if (ZSTR_LEN(arg_name) == 0) {
+            zend_argument_value_error(1, "must not be an empty string");
+            RETURN_THROWS();
+        }
+
+        if (get_parameter_position(fptr, arg_name, num_args) > -1) {
+            RETURN_TRUE;
+        }
+
+        RETURN_FALSE;
+    } else {
+        if (position < 0) {
+            zend_argument_value_error(1, "must be greater than or equal to 0");
+            RETURN_THROWS();
+        }
+
+        RETURN_BOOL(position < num_args);
+    }
+}
+/* }}} */
+
+/* {{{ Returns a parameter specified by its name */
+ZEND_METHOD(ReflectionFunctionAbstract, getParameter)
+{
+    reflection_object *intern;
+    zend_function *fptr;
+    zend_string *arg_name = NULL;
+    zend_long position;
+    struct _zend_arg_info *arg_info;
+    uint32_t num_args;
+    zval reflection;
+
+    ZEND_PARSE_PARAMETERS_START(1, 1)
+        Z_PARAM_STR_OR_LONG(arg_name, position)
+    ZEND_PARSE_PARAMETERS_END();
+
+    GET_REFLECTION_OBJECT_PTR(fptr);
+
+    num_args = fptr->common.num_args;
+    arg_info = fptr->common.arg_info;
+
+    if (fptr->common.fn_flags & ZEND_ACC_VARIADIC) {
+        num_args++;
+    }
+
+    if (num_args < 1) {
+        if (fptr->common.scope) {
+            _DO_THROW_EX("Method %s::%s() has no parameters", ZSTR_VAL(fptr->common.scope->name), ZSTR_VAL(fptr->common.function_name));
+        } else {
+            _DO_THROW_EX("Function %s() has no parameters", ZSTR_VAL(fptr->common.function_name));
+        }
+
+        RETURN_THROWS();
+    }
+
+    if (arg_name != NULL) {
+        if (ZSTR_LEN(arg_name) == 0) {
+            zend_argument_value_error(1, "must not be an empty string");
+            RETURN_THROWS();
+        }
+
+        position = get_parameter_position(fptr, arg_name, num_args);
+
+        if (position == -1) {
+            _DO_THROW("The parameter specified by its name could not be found");
```

Comment:
I think it would help debugging a lot if the parameter name was displayed. WDYT? Similarly, the other error message could include the offset number. Another improvement I'd love to see is to have the function/method name like a few lines above inside the `if (num_args < 1) {` condition.

Reply:
> I think it would help debugging a lot if the parameter name was displayed. WDYT? Similarly, the other error message could include the offset number. Another improvement I'd love to see is to have the function/method name like a few lines above inside the `if (num_args < 1) {` condition.

More than happy to make the exceptions nicer

Reply:
> Another improvement I'd love to see is to have the function/method name like a few lines above inside the `if (num_args < 1) {` condition.

That's already there @kocsismate 

```c
    if (num_args < 1) {
        if (fptr->common.scope) {
            _DO_THROW_EX("Method %s::%s() has no parameters", ZSTR_VAL(fptr->common.scope->name), ZSTR_VAL(fptr->common.function_name));
        } else {
            _DO_THROW_EX("Function %s() has no parameters", ZSTR_VAL(fptr->common.function_name));
        }

        RETURN_THROWS();
    }
```

Reply:
> I think it would help debugging a lot if the parameter name was displayed. WDYT? Similarly, the other error message could include the offset number. Another improvement I'd love to see is to have the function/method name like a few lines above inside the `if (num_args < 1) {` condition.

Off the back of this @kocsismate, do you think it would be worth creating a series of PRs to improve some of the exception messages throughout the language? A lot of them provide little information, or would that constitute a breaking change? I'm not sure where exception messages sit on that.